http_handler.go

```diff
@@ -35,18 +35,19 @@ type errorResponse struct {
 }
 
 type levelHandler struct {
-	logger Logger
+	lvl AtomicLevel
```

Comment:
Why not just add `ServeHTTP` to the `AtomicLevel` type? It doesn't add any overhead if it's not used, and it removes one type and wrapper from this now-crowded package.